scripts/o3de/o3de/compatibility.py

```diff
@@ -0,0 +1,245 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+"""
+This file contains compatibility functions
+"""
+from packaging.version import Version, InvalidVersion
+from packaging.specifiers import SpecifierSet
+import pathlib
+import logging
+from o3de import manifest, utils, cmake
+
+logger = logging.getLogger('o3de.compatibility')
+logging.basicConfig(format=utils.LOG_FORMAT)
+
+def get_project_engine_incompatible_objects(project_path:pathlib.Path, engine_path:pathlib.Path = None) -> set:
+    """
+    Returns any incompatible objects for this project and engine.
+    :param project_path: Path to the project
+    :param engine_path: Optional path to the engine. If not specified, the current engine path is used
+    """
+    # use the specified engine path NOT necessarily engine the project is registered to
+    engine_path = engine_path if engine_path else manifest.get_this_engine_path()
```

Comment:
FYI: You can also do 
```suggestion
    engine_path = engine_path or manifest.get_this_engine_path()
```